### ISSUE TITLE
Use llamacpp or quantized transformers in examples

### ISSUE THREAD

**Issue by cpfiffer** [dottxt-ai/outlines#1140]
The docs currently use `transformers` all over the place, which is very difficult to run if you do not have a massive GPU. I'd like to convert all the examples to use `llamacpp` or a quantized version of `transformers` like so

```python
import outlines
from transformers import BitsAndBytesConfig

model = outlines.models.transformers(
    "microsoft/Phi-3-mini-4k-instruct",
    model_kwargs={
        'quantization_config':BitsAndBytesConfig(
            # Load the model in 4-bit mode
            load_in_4bit=True,

            # Can use 8-bit for more precision
            # load_in_8bit=True,
        )
    }
)
```

We could also use example tabs to show GPU or high-performance production code for those who are interested.

**Comment by sahibpreetsingh12** [comment]
Hi @cpfiffer and @rlouf do we still need to convert all examples to `llamacpp` because I see examples of llamacpp already
If yes any specific examples that are required to be converted to `llamacpp`

**Comment by rlouf** [comment]
Yes, we would like to convert most of cookbooks to `llama.cpp` so most people can run them.
